python/cudf/cudf/core/frame.py

```diff
@@ -235,9 +236,32 @@ def _concat(cls, objs, axis=0, ignore_index=False, sort=False):
                     result_index_length += len(obj)
                     empty_has_index = empty_has_index or len(obj) > 0
 
-        # Get a list of the unique table column names
-        names = [name for f in objs for name in f._column_names]
-        names = OrderedDict.fromkeys(names).keys()
+        if join == "inner":
+            names = [name for obj in objs for name in obj._column_names]
+            names = OrderedDict.fromkeys(names).keys()
+            names_not_in_all = [
+                name
+                for name in names
+                for obj in objs
+                if name not in obj._column_names
+            ]
+            obj = [
+                obj.drop(columns=[x])
+                for obj in objs
+                for x in names_not_in_all
+                if x in obj._column_names
+            ]
```

Comment:
```suggestion
            for obj in objs:
                obj.drop(columns=names_not_in_all, inplace=True, errors="ignore")
```

we could simply do this as `drop` supports ignoring key errors when a column to drop doesn't exist.

Reply:
`objs` are the input objects so we'd need to shallow copy before doing this

Reply:
Ah, yes. Missed it totally, Corrected the suggestion 👍

Reply:
Great! Thanks, I've updated this :D